src/check_deps.rs

```diff
@@ -0,0 +1,97 @@
+use crate::utils::rename_deps;
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::Path;
+use toml_edit::DocumentMut;
+pub fn check_dependencies(
+    cargo_toml_path: &Path,
+    crates_versions: &BTreeMap<String, String>,
+    overwrite: bool,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let _ = check_dependencies_impl(cargo_toml_path, crates_versions, overwrite)?;
```

Comment:
nit

```suggestion
    check_dependencies_impl(cargo_toml_path, crates_versions, overwrite)?;
```

Reply:
`let _ =` is an antipattern! Don't ever write code like that imo